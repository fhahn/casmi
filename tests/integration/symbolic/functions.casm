// symbolic

function (symbolic) foo : Int -> Int
function (symbolic) bar : Int -> Int
function baz : Int -> Int
function quux : Int -> Int

rule main = {
	
	assert ( symbolic (foo(1)) )
	assert ( symbolic (bar(23)) )
	assert ( foo(1) != undef )
	assert ( bar(23) != undef )

	assert ( baz(1) = undef )
	assert ( quux(5) = undef )
	assert ( not symbolic( baz(1) ) )

	program(self) := undef
	
}

init main

