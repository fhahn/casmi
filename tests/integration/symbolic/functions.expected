forklog:
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,stfoo(1,1,sym2)).%CREATE: foo(1)
tff(symbolNext, type, sym3: $int).
fof(id1,hypothesis,stbar(1,23,sym3)).%CREATE: bar(23)
fof(id2,hypothesis,stfoo(2,1,sym2)).%SYMBOLIC: foo(1)
fof(id3,hypothesis,stbar(2,23,sym3)).%SYMBOLIC: bar(23)
fof(final0,hypothesis,stfoo(0,1,sym2)).%FINAL: foo(1)
fof(final1,hypothesis,stbar(0,23,sym3)).%FINAL: bar(23)

